bgpd/bgp_evpn.c

```diff
@@ -3537,7 +3537,7 @@ static int process_type5_route(struct peer *peer, afi_t afi, safi_t safi,
 
 	if (attr) {
 		is_valid_update = true;
-		if (is_zero_mac(&attr->rmac) && is_zero_esi(&evpn.eth_s_id) &&
+		if (is_zero_mac(&attr->rmac) &&
```

Comment:
ESI is a dead field in type-5 routes. It is always zero.

i.e. it is sufficient to check of the RMAC and GW-IP are zero to treat as a withdraw.

Reply:
Hi Anuradha, is ESI not going to be used as overlay index at all?

Reply:
No. In the past it was part of the overlay-index code and compared but never actually parsed/populated. So it was dead code for all practical purposes; taking up space and processing without serving a purpose.

Now ESI is processed, in type-1/type-2/type-4 routes and is in an inline EVPN route attr. I wanted to avoid confusion with having the same param-name in two different places (the code is already complex enough!). That was the main reason for the overlay index cleanup.